### ISSUE TITLE
Is there a way to delete states by identifier?

### ISSUE THREAD

**Issue by rcheetah** [icebaker/ruby-nano-bots#19]
The state names in the storage folder are encrypted, as well as the contents. I wonder if it is possible to delete a given state by providing the state string? Something like `nb delete state123`

**Comment by icebaker** [comment]
Hi! Not through the command line, but what you can do is get details about the state:

```
nb - ABC state
```

It will return something like:
```ruby
{:state=>
  {:path=>
    "/home/user/.local/state/nano-bots/ruby-nano-bots/none/unknown/0-0-0/XPTO/state.json",
   :content=>
    {:key=>"ABC",
     :history=>
      [{:at=>"2024-06-04 17:05:58 -0300", :who=>"user", :mode=>"eval", :input=>"hi", :message=>"hi"},
       {:who=>"AI", :message=>"Hello! How can I assist you today?", :mode=>"eval", :output=>"Hello! How can I assist you today?\n", :at=>"2024-06-04 17:06:01 -0300"}]}}}
```

Then you can delete the file:
```text
/home/user/.local/state/nano-bots/ruby-nano-bots/none/unknown/0-0-0/XPTO/state.json
```

**Comment by rcheetah** [comment]
Perfect, thank you very much! This works for my cause. I didn't see the debugging part in the readme (mostly because I was searching for 'delete' or 'state').

The only thing that bothers me: Debugging a non-existing state will create it. So if I want to check if a state with a given name exists, I will always create it, and therefore it will exists. Is this intended behaviour? Seems like debugging a non-existing state should throw an error.

**Comment by icebaker** [comment]
@rcheetah  Yeah, you would need to check the state file, and if it's empty (`history` array empty), it means it "doesn't exist." This wasn't intentionally built that way. The reason it happens is:

When CLI is used through `nb` and it's a command related to a "bot," it will [create the bot](https://github.com/icebaker/ruby-nano-bots/blob/v3.2.0/controllers/interfaces/cli.rb#L72) before executing the command.

In this process of creating the bot, it will call [`build_path_and_ensure_state_file!`](https://github.com/icebaker/ruby-nano-bots/blob/main/controllers/session.rb#L37), thereby creating the file. The goal was to ensure that before the bot executes, if it should have a state, the state file is writable and ready to be filled.

The unintended collateral effect is creating a new state file with an empty history for the ones that don't exist when just checking/debugging.

**Comment by icebaker** [comment]
@rcheetah I'm not sure about throwing an error, but in [version 3.3.0](https://github.com/icebaker/ruby-nano-bots/releases/tag/v3.3.0) and beyond, it will no longer create a file if the state does not exist and will return `nil` instead:

```bash
nb - XPTO state
nil
```

**Comment by palladius** [comment]
Can we add a functionality like:

```
nb - ABC delete-state
```
?

**Comment by rcheetah** [comment]
@icebaker Awesome, thank you!
